brion/plugin/compartmentReportDummy.cpp

```diff
@@ -0,0 +1,170 @@
+    return dunit;
+}
+
+const std::string& CompartmentReportDummy::getTimeUnit() const
+{
+    static const std::string tunit = "lightmilliseconds";
+    return tunit;
+}
+
+float CompartmentReportDummy::getStartTime() const
+{
+    return 0.f;
+}
+
+float CompartmentReportDummy::getEndTime() const
+{
+    return 1024.f;
+}
+
+float CompartmentReportDummy::getTimestep() const
+{
+    return 1.f;
+}
+
+void CompartmentReportDummy::updateMapping( const GIDSet& gids )
+{
+    if( _gids == gids && !gids.empty( ))
+        return;
+
+    std::random_device dev;
+    std::mt19937_64 engine( dev( ));
+    std::uniform_int_distribution < uint32_t > distribution;
+    engine.seed( _defaultGIDs ); // reproducible randomness, please
+
+    GIDSet all;
+    while( all.size() < _defaultGIDs )
+        all.insert( distribution( engine ));
+
+    const auto& subset = gids.empty() ? all : gids;
+    _gids = _computeIntersection( all, subset );
+
+    _counts.clear();
+    _offsets.clear();
+
+    // aim for ~1k compartments/cell
+    static const size_t spread = 0x1f; // mask, 31
+    static const size_t maxCompartments = 256 /* x 4B/value */ - spread / 2;
+    uint64_t offset = 0;
+
+    while( _counts.size() < _gids.size() - 1 )
+    {
+        _counts.push_back( uint16_ts( ));
+        _offsets.push_back( uint64_ts( ));
+
+        for( size_t nCompartments = 0; nCompartments < maxCompartments;
+             nCompartments += _counts.back().back( ))
+        {
+            _counts.back().push_back(
+                uint16_t( distribution( engine ) & spread ));
+            _offsets.back().push_back( offset );
+            offset += _counts.back().back();
+        }
+    }
+
+    // Last cell: fill up to desired size to correct random errors above
+    const size_t targetSize = _gids.size() * 256 /* x 4B/value */;
+    _counts.push_back( uint16_ts( ));
+    _offsets.push_back( uint64_ts( ));
+    while( offset < targetSize )
+    {
+        _counts.back().push_back(
+            uint16_t( distribution( engine ) & spread ));
+        _offsets.back().push_back( offset );
+        offset += _counts.back().back();
+    }
+
+    // correct last entry
+    _counts.back().back() -= (offset - targetSize);
+    _offsets.back().back() -= (offset - targetSize);
+}
+
+size_t CompartmentReportDummy::getFrameSize() const
+{
+    size_t frameSize = 0;
+    for( const auto& counts : _counts )
+        frameSize += std::accumulate( counts.begin(), counts.end(), 0 );
+    return frameSize;
+}
+
+floatsPtr CompartmentReportDummy::loadFrame( const float time ) const
+{
+    floatsPtr buffer( new floats( getFrameSize( )));
+    if( _randomValues )
+    {
+        std::random_device dev;
+        std::mt19937_64 engine( dev( ));
+        std::uniform_int_distribution < int16_t > distribution;
+        engine.seed( unsigned( time )); // reproducible randomness, please
+
+        for( float& value : *buffer )
+            value = float( distribution( engine )) / 1000.f;
```

Comment:
Maybe it would be good to have the floating point numbers generated in a user defined range. At least it should be documented which is the default range (which looks a bit strange I have to say).